### PR TITLE
Fix matching of opening curly quote in description

### PR DESCRIPTION
**Pull request by kdeldycke**
Without this fix, an item of the form:

```markdown
- [What Google Learned From Its Quest to Build the Perfect Team](https://www.nytimes.com/2016/02/28/magazine/what-google-learned-from-its-quest-to-build-the-perfect-team.html) - “Google's data indicated that psychological safety, more than anything else, was critical to making a team work. (…) The behaviors that create psychological safety — conversational turn-taking and empathy — are part of the same unwritten rules we often turn to, as individuals, when we need to establish a bond.”
```

gets reported under the following linting error:

```shell-session
  ✖  105:176  List item description must start with valid casing            remark-lint:awesome-list-item
```

See for instance: https://github.com/kdeldycke/awesome-management/runs/1291575555?check_suite_focus=true

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you add a fixture so this is tested?

https://github.com/sindresorhus/awesome-lint/blob/master/test/fixtures/list-item/0.md

**Comment by kdeldycke** [comment]
You're right @sindresorhus and I'm sorry, I was a little lazy on that one. Just added a fixture to catch this issue.

With the fix:

```shell-session
❯ npm test

> awesome-lint@0.16.0 test
> xo && ava

  (...)

  1 test failed

  rules › list-item › list-item - valid

  /Users/kde/awesome-lint/test/rules/list-item.js:13

   12:   const messages = await lint({config, filename: 'test/fixtures/list-item/0.md'});
   13:   t.deepEqual(messages, []);                                                      
   14: });                                                                               

  Difference:

    [
  -   Object { … },
    ]

  test/rules/list-item.js:13:4
  processTicksAndRejections (node:internal/process/task_queues:93:5)

npm ERR! code 1
npm ERR! path /Users/kde/awesome-lint
npm ERR! command failed
npm ERR! command sh -c xo && ava

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kde/.npm/_logs/2020-10-26T10_03_55_156Z-debug.log
```

After the fix:
```shell-sessioin
❯ npm test

> awesome-lint@0.16.0 test
> xo && ava

  (...)

 
  65 tests passed
```

File: rules/list-item.js [hunk 1]
```diff
@@ -250,7 +250,7 @@ function validateListItemPrefixCasing(prefix, file) {
 	}
 
 	if (!listItemPrefixCaseWhitelist.has(caseOf(firstWord))) {
-		if (!/\d/.test(firstWord) && !/^["'(]/.test(firstWord)) {
+		if (!/\d/.test(firstWord) && !/^["“'(]/.test(firstWord)) {
```
Comment (sindresorhus) [inline rules/list-item.js:253]: Note that the left and right quotes are separate Unicode characters. `“”`

Comment (kdeldycke) [reply rules/list-item.js:253]: Yes, that normal. We should not allow for a description to starts with a closing `”` curved quote. I added a test case for that at: https://github.com/sindresorhus/awesome-lint/pull/127/commits/e318783dc7520afc227816f8f43f67b8d3be7c8f

**Comment by sindresorhus** [comment]
Thanks :)

**Comment by kdeldycke** [comment]
Thanks @sindresorhus for the merge! :)

**Pull request merged**